### DISCUSSION TITLE
Adding a nullable foreign key user_id to the statuses table

### DISCUSSION THREAD

**Discussion by JospyK** [spatie/laravel-model-status#86] [Ideas]
Hi, hope you're doing good.
I thought it would be nice to add a **nullable foreign key _user_id_ to the _statuses table_**. Sometimes it is needed to store the id of the user related to the status change. What do you think?

**Comment by yarscript** [comment]
Hi there!
@JospyK I think it would be better to have polymorphic relation kind of `author` or `changed_by`. I did custom status package includes this thing because I really need this feature in my projects
Thanks!

**Reply by yarscript** [reply]
Live the link here as an example of implementation:
https://github.com/yarscript/laravel-status
